Helper/CheckboxesHandler.php

```diff
@@ -0,0 +1,107 @@
+<?php
+/**
+ * Bolt magento2 plugin
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * http://opensource.org/licenses/osl-3.0.php
+ *
+ * @category   Bolt
+ * @package    Bolt_Boltpay
+ * @copyright  Copyright (c) 2018 Bolt Financial, Inc (https://www.bolt.com)
+ * @license    http://opensource.org/licenses/osl-3.0.php  Open Software License (OSL 3.0)
+ */
+
+namespace Bolt\Boltpay\Helper;
+
+use Magento\Framework\App\Helper\AbstractHelper;
+use Magento\Framework\App\Helper\Context;
+use Magento\Newsletter\Model\SubscriberFactory;
+
+/**
+ * Boltpay web hook helper
+ *
+ * @SuppressWarnings(PHPMD.TooManyFields)
+ * @SuppressWarnings(PHPMD.CouplingBetweenObjects)
+ */
+class CheckboxesHandler extends AbstractHelper
+{
+    const CATEGORY_NEWSLETTER = 'NEWSLETTER';
+    const COMMENT_PREFIX_TEXT = 'BOLTPAY INFO :: checkboxes';
+
+    /**
+     * @var SubscriberFactory
+     */
+    private $subscriberFactory;
+
+    /**
+     * @var Bugsnag
+     */
+    private $bugsnag;
+
+    /**
+     * @param Context $context
+     * @param OrderHelper $orderHelper
+     * @param Bugsnag $bugsnag
+     */
+    public function __construct(
+        Context $context,
+        Bugsnag $bugsnag,
+        SubscriberFactory $subscriberFactory
+    )
+    {
+        parent::__construct($context);
+        $this->subscriberFactory = $subscriberFactory;
+        $this->bugsnag = $bugsnag;
+    }
+
+    /**
+     * Handle checkboxes
+     *
+     * @param OrderModel $order
+     * @param array $checkboxes
+     */
+    public function handle($order, $checkboxes)
+    {
+        $comment = '';
+        $needSubscribe = false;
+        foreach ($checkboxes as $checkbox) {
+            if ($checkbox['category'] == self::CATEGORY_NEWSLETTER && $checkbox['value']) {
```

Comment:
```
  "checkboxes": [
    {
      "checkbox_id": "subscription",
      "text": "Sign me up for the newsletter!",
      "value": true,
      "category": "newsletter",
      "features": [
        "subscribe_to_platform_newsletter"
      ]
    }
  ]
```

To set `needSubscribe` to `true` we should also ensure that the `subscribe_to_platform_newsletter` feature is present in the features field.

Something like `$checkbox['features'] && in_array("subscribe_to_platform_newsletter", $checkbox['features'])`


Apart from this change, everything else looks good to me!

Reply:
Done